src/arch/xtensa/smp/idc.c

```diff
@@ -299,15 +328,18 @@ static uint32_t idc_get_done_bit_mask(int core)
 /**
  * \brief Initializes IDC data and registers for interrupt.
  */
-void arch_idc_init(void)
+int arch_idc_init(void)
 {
 	int core = arch_cpu_get_id();
+	int ret;
 
 	trace_idc("arch_idc_init()");
 
 	/* initialize idc data */
 	struct idc **idc = idc_get();
 	*idc = rzalloc(RZONE_SYS, SOF_MEM_CAPS_RAM, sizeof(**idc));
+	if (!*idc)
```

Comment:
`rmalloc_sys()` panics if there is no free memory to allocate which practically makes the `if(null)` a dead code in every place that follows an allocation from `RZONE_SYS`, and what is more, it suggests the reader that the allocation may really fail at run-time being handled by the callers. If not sure why KW is unable to analyze that path and follow `switch (zone & RZONE_TYPE_MASK)` correctly.